src/net_processing.cpp

```diff
@@ -1204,15 +1204,6 @@ bool static AlreadyHave(const CInv& inv) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
     return true;
 }
 
-static void RelayTransaction(const CTransaction& tx, CConnman* connman)
-{
-    CInv inv(MSG_TX, tx.GetHash());
```

Comment:
I'd prefer if that inv construction was kept in a function called `RelayTransaction`, where we pass in the tx(id). The caller should not be bothered dealing with inv-internals: E.g. that we can only push invs of type TX or BLOCK.

Reply:
Sounds good - I also split `CNode::PushInventory` into block and transaction variants for the same reason - all the callers were explicit and it simplified the callers.